apiserver/common/crossmodel/auth.go

```diff
@@ -344,10 +365,20 @@ func (a *authenticator) checkMacaroons(
 	}
 	declared := checkers.InferDeclared(coremacaroon.MacaroonNamespace, mac)
 	authlogger.Debugf("check macaroons with declared attrs: %v", declared)
+
+	if a.ctxt.jaasOfferBakery != nil {
+		// We only need to inject relationKey for jaas flow.
+		if _, ok := declared[relationKey]; !ok {
+			if relation, ok := requiredValues[relationKey]; ok {
+				declared[relationKey] = relation
+			}
+		}
+	}
 	username, ok := declared[usernameKey]
 	if !ok {
 		return nil, apiservererrors.ErrPerm
 	}
+	userTag := names.NewUserTag(username)
```

Comment:
Internally it's better to try and avoid using tag if possible. Can we just pass username to the create macaroon methods below.